### PR TITLE
fix an rrule bug that causes improper HOURLY/MINUTELY calculation

### PR DESCRIPTION
**Pull request by ryanpetrello**
see: https://github.com/ansible/awx/issues/8071

### CODE REVIEW

File: awx/main/models/schedules.py [hunk 1]
```diff
@@ -205,10 +205,15 @@ def rrulestr(cls, rrule, fast_forward=True, **kwargs):
                     'A valid TZID must be provided (e.g., America/New_York)'
                 )
 
-        if fast_forward and ('MINUTELY' in rrule or 'HOURLY' in rrule):
+        if (
+            fast_forward and
+            ('MINUTELY' in rrule or 'HOURLY' in rrule) and
+            'COUNT=' not in rrule
```
Comment (ryanpetrello) [inline awx/main/models/schedules.py:211]: We really only need this optimization if people set up *unbounded* per minute/hour rules in the past.

File: awx/main/models/schedules.py [hunk 2]
```diff
@@ -205,10 +205,15 @@ def rrulestr(cls, rrule, fast_forward=True, **kwargs):
...
+        ):
             try:
                 first_event = x[0]
-                if first_event < now():
+                # If the first event was over a week ago...
+                if (first_event - now()).days > 7:
```
Comment (ryanpetrello) [inline awx/main/models/schedules.py:216]: Similarly, if the first per-minute/per-hour event *is actually in the past week*, it's not going to be expensive, so we can skip this optimization.

**Comment by jakemcdermott** [comment]
arrrr

**Review by AlanCoding — APPROVED** [review]

**Review by jakemcdermott — APPROVED** [review]

File: awx/main/tests/functional/models/test_schedule.py [hunk 3]
```diff
@@ -161,6 +161,57 @@ def test_computed_fields_turning_off_by_deleting(self, job_template):
...
+    first_event = sched.rrulestr(sched.rrule)[0]
+    assert first_event.replace(tzinfo=None) == recent
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('freq, delta', (
+    ('MINUTELY', 1),
+    ('HOURLY', 1)
+))
+def test_really_old_dtstart(job_template, freq, delta):
```
Comment (jakemcdermott) [inline awx/main/tests/functional/models/test_schedule.py:189]: lgtm

File: awx/main/models/schedules.py [hunk 4]
```diff
@@ -205,10 +205,15 @@ def rrulestr(cls, rrule, fast_forward=True, **kwargs):
...
+        if (
+            fast_forward and
+            ('MINUTELY' in rrule or 'HOURLY' in rrule) and
+            'COUNT=' not in rrule
+        ):
             try:
                 first_event = x[0]
-                if first_event < now():
+                # If the first event was over a week ago...
+                if (now() - first_event).days > 7:
```
Comment (ryanpetrello) [inline awx/main/models/schedules.py:216]: Similarly, if the first per-minute/per-hour event is actually _in the past week_, it's not going to be expensive to burst out the rrule, so we can skip this optimization.

**Pull request merged**